src/Common/src/System/Net/WebSockets/ManagedWebSocket.cs

```diff
@@ -279,7 +279,8 @@ public override Task SendAsync(ArraySegment<byte> buffer, WebSocketMessageType m
 
             try
             {
-               WebSocketValidate.ThrowIfInvalidState(_state, _disposed, s_validSendStates);
+                ThrowIfDisposed();
```

Comment:
ManagedWebSocket was originally written for use with ClientWebSocket, which already does such a disposal check:
https://github.com/dotnet/corefx/blob/master/src/System.Net.WebSockets.Client/src/System/Net/WebSockets/ClientWebSocket.cs#L164
And it's also used by ASP.NET.  Is there a way to avoid paying for the double check?  Does adding this here cause any problems for ASP.NET? Etc.